custom_components/tahoma/climate_devices/atlantic_electrical_heater.py

```diff
@@ -60,17 +65,20 @@
     PRESET_ECO: PRESET_STATE_ECO,
     PRESET_FROST_PROTECTION: PRESET_STATE_FROST_PROTECTION,
     PRESET_NONE: PRESET_STATE_OFF,
+    PRESET_AUTO: PRESET_STATE_AUTO,
+    PRESET_PROG: PRESET_STATE_PROG,
 }
 
 TAHOMA_TO_PRESET_MODE = {v: k for k, v in PRESET_MODE_TO_TAHOMA.items()}
 
 # Map TaHoma HVAC modes to Home Assistant HVAC modes
 TAHOMA_TO_HVAC_MODE = {
+    "on": HVAC_MODE_HEAT,
```

Comment:
Is 'on' the preferred (and supported) state for all devices? Having it as the top choice for `HVAC_MODE_HEAT` will mean that it is selected as state for `set_hvac_mode`, instead of `basic`.

Reply:
According to my tests, the "HVAC_MODE_TO_TAHOMA" list will keep the last entry of the "TAHOMA_TO_HVAC_MODE" list.

So in fact we have the following results : 
HVAC_MODE_HEAT:"basic",
HVAC_MODE_OFF:"standby", 
HVAC_MODE_AUTO:"internal" ,

It will not work for the PASS_ACTUATOR device but it's not very usefull, just the PRESET is usefull (it's not really a heater).

Reply:
Ok! It seems that they are more different than I thought, thus I will split them in two different implementations.

https://github.com/iMicknl/ha-tahoma/pull/454 <-- This one will hopefully fix your issues as well. Do we need to add the other presets here as well?